plugins/processors/ec2tagger/constants.go

```diff
@@ -0,0 +1,93 @@
+  ##
+  ## Specify which tag to use to get the specified disk device name from input Metric
+  # disk_device_tag_key = "device"
+  ##
+  ## Amazon Credentials
+  ## Credentials are loaded in the following order
+  ## 1) Assumed credentials via STS if role_arn is specified
+  ## 2) explicit credentials from 'access_key' and 'secret_key'
+  ## 3) shared profile from 'profile'
+  ## 4) environment variables
+  ## 5) shared credentials file
+  ## 6) EC2 Instance Profile
+  # access_key = ""
+  # secret_key = ""
+  # token = ""
+  # role_arn = ""
+  # profile = ""
+  # shared_credential_file = ""
+`
+
+const (
+	ec2InstanceTagKeyASG = "aws:autoscaling:groupName"
+	cwDimensionASG       = "AutoScalingGroupName"
+	mdKeyInstanceId      = "InstanceId"
+	mdKeyImageId         = "ImageId"
+	mdKeyInstaneType     = "InstanceType"
+	ebsVolumeId          = "EBSVolumeId"
+)
+
+const (
+	metadataCheckStrStartInitialization         = "ec2tagger: EC2 IMDS has started initialization."
+	metadataCheckStrTagNotSupported             = "ec2tagger: Unsupported EC2 Metadata key: %s."
+	metadataCheckStrInstanceDocumentFailure     = "ec2tagger: Unable to retrieve Instance Metadata Tags."
+	metadataCheckStrEC2InstanceTagger           = "This plugin must only be used on an EC2 instance."
+	metadataCheckStrIncreaseHopLimit            = "ec2tagger: Please increase hop limit to 3 by following this document https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/configuring-instance-metadata-options.html#configuring-IMDS-existing-instances."
+	ec2TagAndVolumeCheckStrInitRetrievalSuccess = "ec2tagger: Initial retrieval of tags succeeded"
+	ec2TagAndVolumeCheckStrStartInitialization  = "ec2tagger: EC2 tagger has started initialization."
+	ec2VolumeCheckStrInitRetrievalFailure       = "ec2tagger: Unable to describe ec2 volume for initial retrieval: %v"
+	ec2TagCheckStrInitRetrievalFailure          = "ec2tagger: Unable to describe ec2 tags for initial retrieval: %v"
+	ec2TagAndVolumeCheckStrStartRefresh         = "ec2tagger refreshing: EC2InstanceTags needed %v, retrieved: %v, ebs device needed %v, retrieved: %v"
+	ec2TagAndVolumeCheckStrStopRefresh          = "ec2tagger: Refresh is no longer needed, stop refreshTicker."
+	ec2TagAndVolumeCheckStrRetryFailure         = "ec2tagger: %v retry initial retrieval of tags and volumes"
+	ec2VolumeCheckStrRefreshFailure             = "ec2tagger: Error refreshing EC2 volumes, keeping old values : %+v"
+	ec2TagCheckStrRefreshFailure                = "ec2tagger: Error refreshing EC2 tags, keeping old values : %+v"
+	ec2TagAndVolumeCheckStrSuccess              = "ec2tagger: Finished initial retrieval of tags and volumes"
+)
+
+var (
+	defaultRefreshInterval = 180 * time.Second
+	// backoff retry for ec2 describe instances API call. Assuming the throttle limit is 20 per second. 10 mins allow 12000 API calls.
+	backoffSleepArray = []time.Duration{0, 1 * time.Minute, 1 * time.Minute, 3 * time.Minute, 3 * time.Minute, 3 * time.Minute, 10 * time.Minute}
```

Comment:
Same idea initially. I also intend to re-structure the ec2 tagger interface but consider its out of scope; therefore, we could leave it in another PR.

Reply:
How is it out of scope? The purpose of the task is to reduce time to reach IMDS. This is relevant.

Reply:
It does not relate since `backoffSleepArray` is for [ec2 API](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR298-R307) which uses describeTag and describe Volume which does not relate to IMDS while `imds` using [ec2metadata](https://github.com/aws/amazon-cloudwatch-agent/pull/480/files#diff-6638b992018408bf44cac12361171fd050ff44fcc24c28f326988f4768a6dc2dR255)

Reply:
For more information on these two:
* EC2Metadata: https://github.com/aws/aws-sdk-go/blob/main/aws/ec2metadata/api.go
* EC2: https://github.com/aws/aws-sdk-go/tree/main/service/ec2